shared/actions/config/index.js

```diff
@@ -30,6 +30,28 @@ export function getConfig (): AsyncAction {
   }
 }
 
+export function getExtendedConfig (): AsyncAction {
+  return dispatch => {
+    return new Promise((resolve, reject) => {
+      const params : config_getExtendedStatus_rpc = {
+        method: 'config.getExtendedStatus',
+        param: {},
+        incomingCallMap: {},
+        callback: (error, extendedConfig) => {
+          if (error) {
+            reject(error)
+          }
+
+          dispatch({type: Constants.extendedConfigLoaded, payload: {extendedConfig}})
+          resolve(extendedConfig)
```

Comment:
Usually we wouldn't return anything in this resolve, but because this is happening on the remote store, the remote store won't get the latest state when this finishes it's dispatching.

This is a deeper subtlety that might lead to bugs where we expect dispatches to be synchronous (or chainable if async) when working in remote components. We should think about a nice solution for this.